### PR TITLE
Feature/add upgradability

### PR DESCRIPTION
**Pull request by invocamanman**
- Add upgradabiltiy
- Update deployments
- Add permit function

### CODE REVIEW

File: contracts/Bridge.sol [hunk 1]
```diff
@@ -371,4 +384,77 @@ contract Bridge is Ownable, DepositContract {
...
+     */
+    function _permit(
+        address token,
+        uint256 amount,
+        bytes calldata permitData
+    ) internal {
+        bytes4 sig = _getSelector(permitData);
+        require(
+            sig == _PERMIT_SIGNATURE,
+            "HezMaticMerge::_permit: NOT_VALID_CALL"
```
Comment (ignasirv) [inline contracts/Bridge.sol:416]: Change HezMaticMerge to Bridge

File: test/contracts/helpers/permit-helper.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+const {
+    ethers,
+} = require('hardhat');
+const { expect } = require('chai');
+
+async function createPermitSignature(tokenContractInstance, wallet, spenderAddress, value, nonce, deadline) {
```
Comment (krlosMata) [inline test/contracts/helpers/permit-helper.js:6]: Add comments

File: test/contracts/helpers/permit-helper.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    const rawSignature = await wallet._signTypedData(domain, types, values);
+    const signature = ethers.utils.splitSignature(rawSignature);
+    const recoveredAddressTyped = ethers.utils.verifyTypedData(domain, types, values, rawSignature);
+    expect(recoveredAddressTyped).to.be.equal(wallet.address);
+
+    return signature;
+}
+
+const ifacePermit = new ethers.utils.Interface(['function permit(address,address,uint256,uint256,uint8,bytes32,bytes32)']);
```
Comment (krlosMata) [inline test/contracts/helpers/permit-helper.js:46]: Add comments

File: package.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
```
Comment (krlosMata) [inline package.json:1]: If `permit-helper.js` is meant to be used by third-parties I would include it in the `index.js`

File: docker/Dockerfile.geth [hunk 5]
```diff
@@ -1,4 +1,4 @@
-FROM ethereum/client-go:v1.10.15
+FROM ethereum/client-go
```
Comment (krlosMata) [inline docker/Dockerfile.geth:1]: We decided to hardcode a version for `ethereum/client-go` in order to have always the same deployment and to not be dependent on changes in `ethereum/client-go` versions. This has been a source of issues in the past.
Therefore, I would setup a concrete version (to decode with `zkNode` team).
If needed, the version will be updated.

**Review by krlosMata — CHANGES_REQUESTED** [review]

**Review by krlosMata — APPROVED** [review]
:smile_cat:

**Review by laisolizq — APPROVED** [review]
:smile_cat:

**Pull request merged**